### ISSUE TITLE
Warn on `echo "$var"` when `$var` might be `-e`

### ISSUE THREAD

**Issue by andersk** [koalaman/shellcheck#2674]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

var="-e"
echo "$var"
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

ShellCheck should warn that `echo "$var"` produces non-portable results: the value might unintentionally be parsed as an `echo` option depending on the shell.

```console
$ bash test.sh

$ dash test.sh
-e
```

`printf '%s\n' "$var"` should be suggested as an alternative.

This is a sort of converse to [SC3037](https://www.shellcheck.net/wiki/SC3037), which is for code that *intentionally* tries to use a non-portable `echo` option.

(Inspired by [bash’s echo command is broken](https://www.youtube.com/watch?v=lq98MM2ogBk), which shows a real-world example of this bug in the `zipgrep` shell script.)
